pypeit/find_objects.py

```diff
@@ -561,50 +613,60 @@ def global_skysub(self, skymask=None, update_crmask=True, trim_edg=(3,3),
         # Return
         return global_sky
 
-    def load_skyregions(self, skymask_init):
+    def load_skyregions(self, sky_region_file=None):
         """
-        Load or generate the sky regions
+        Generate or load the sky regions.
+
+        Preference is given to generate the sky regions using the "user_regions" argument,
+        since this is set by the user in their .pypeit file, in the following way:
+
+        .. code-block:: ini
+
+            [reduce]
+                [[skysub]]
+                    user_regions = :25,75:
+
+        The alternative is to load a MasterSkyRegions file (generated by the pypeit_skysub_regions GUI). If the
+        user wishes to load the MasterSkyRegions file, they need to set user_regions = True:
+
+        .. code-block:: ini
+
+            [reduce]
+                [[skysub]]
+                    user_regions = master
 
         Parameters
         ----------
-        skymask_init :  `numpy.ndarray`_
-            A boolean array of sky pixels (True is pixel is a sky region)
+        sky_region_file : str, None, optional
+            Name of the sky regions file
 
         Returns
         -------
-        skymask_init :  `numpy.ndarray`_
-            A boolean array of sky pixels (True is pixel is a sky region)
-        usersky : bool
-            If the user has defined the sky, set this variable to True (otherwise False).
+        skymask :  `numpy.ndarray`_
+            A boolean array of sky pixels (True is a pixel that corresponds to a sky region)
         """
-        usersky = False
-        if self.par['reduce']['skysub']['load_mask']:
-            # Check if a master Sky Regions file exists for this science frame
-            file_base = os.path.basename(self.sciImg.files[0])
-            prefix = os.path.splitext(file_base)
-            if prefix[1] == ".gz":
-                sciName = os.path.splitext(prefix[0])[0]
-            else:
-                sciName = prefix[0]
-
-            # Setup the master frame name
-            master_dir = self.caliBrate.master_dir
-            master_key = self.caliBrate.fitstbl.master_key(0, det=self.det) + "_" + sciName
-
-            regfile = masterframe.construct_file_name(buildimage.SkyRegions,
-                                                      master_key=master_key,
-                                                      master_dir=master_dir)
+        user_regions = self.par['reduce']['skysub']['user_regions']
+        # Perform some checks
+        if user_regions is None:
```

Comment:
What if `user_regions == ''` ?

Reply:
@rcooke-ast can you please comment.

Reply:
This possibility gets captured on line 346